src/components/icon/icon.md

```diff
@@ -1,44 +1,64 @@
-The `<Icon />` component's API is mostly self-evident and you can refer to the props table at the bottom of this page.
-However, let's look at the color properties in more detail.
-We use three different properties to apply different colors to an SVG:
+The `<Icon>` component represents our latest patterns with using SVGs.
+It is also designed to be accessible -- please read our accessibility section below!
 
-- `fill` sets the fill color
-- `stroke` sets the stroke color
-- `currentColor` is an SVG trick to provide an additional `fill` and/or `stroke` color
+Forward facing note:
 
-Basic Usage
+- `currentColor` API is being deprecated in favor of self-contained SVGs -- please use `skip` for now
+- `fill` API is being deprecated in favor of self-contained SVGs -- please use `skip` for now
+- `skip` API is being deprecated in favor of self-contained SVGs -- please use `skip` for now
+- *Self-contained are SVGs with all coloring styling applied in the SVG file
+
+### Basic Usage
 
 ```jsx
 const icon = require('../../svgs/icon-caution-fill.svg').default;
-<Icon name={icon.id} size="large" currentColor="caution" title="alert icon" />
+<Icon name={icon.id} size="large" currentColor="skip" fill="skip" stroke="skip" title="alert icon" />
 ```
-Sizes
+
+### Sizes
+
+For the most part, the `size` API is the main configurable property.
 
 ```jsx
-const styles = require('./example.css'); /* For example purposes only! */
 const icon = require('../../svgs/icon-caution-fill.svg').default;
+<Icon name={icon.id} size="small" currentColor="skip" fill="skip" stroke="skip" />
+```
 
-<div>
-  <span className={styles['icon-example']}>
-    <Icon name={icon.id} size="small" currentColor="caution" />
-  </span>
-  <span className={styles['icon-example']}>
-    <Icon name={icon.id} size="medium" currentColor="caution" />
-  </span>
-  <span className={styles['icon-example']}>
-    <Icon name={icon.id} size="large" currentColor="caution" />
-  </span>
-</div>
+```jsx
+const icon = require('../../svgs/icon-caution-fill.svg').default;
+<Icon name={icon.id} size="medium" currentColor="skip" fill="skip" stroke="skip" />
 ```
 
-Current Color Example
+```jsx
+const icon = require('../../svgs/icon-caution-fill.svg').default;
+<Icon name={icon.id} size="large" currentColor="skip" fill="skip" stroke="skip" />
+```
 
-Can be one of 'primary', 'action', 'highlight', or 'caution';
+### Accessibility
```

Comment:
Loving all the new Accessibility documentation! 🥳

Reply:
Yeah, after looking at our component doc page, I think accessibility should be the first section of every documentation page:

- highlights how to interact with the component (even for basic examples!)
- highlights accessibility features on top of that 
- makes sure we are doing the research on this